src/v/cluster/rm_stm.cc

```diff
@@ -1782,8 +1782,11 @@ void rm_stm::apply_fence(model::record_batch&& b) {
 
     auto [fence_it, _] = _log_state.fence_pid_epoch.try_emplace(
       bid.pid.get_id(), bid.pid.get_epoch());
-    if (fence_it->second < bid.pid.get_epoch()) {
+    if (fence_it->second <= bid.pid.get_epoch()) {
```

Comment:
> cluster: Add tx_seqs map to log_state
Store seq number in map inside log state

The commit message explains that we are tracking the sequence number and the code reflects that, but neither the commit message nor a comment in the commit explains this change related to comparing fence/epoch. Is it fixing an unrelated bug? How would a future developer looking at this commit reason about this change to the inequality?

Reply:
added a comment and updated a commit message